src/app/filelogger.cpp

```diff
@@ -176,7 +176,7 @@ void FileLogger::flushLog()
 void FileLogger::openLogFile()
 {
     if (!m_logFile.open(QIODevice::WriteOnly | QIODevice::Append | QIODevice::Text)
-        || !m_logFile.setPermissions(QFile::ReadOwner | QFile::WriteOwner))
+        || (!(m_logFile.permissions() & (QFile::ReadOwner | QFile::WriteOwner)) && !m_logFile.setPermissions(QFile::ReadOwner | QFile::WriteOwner)))
```

Comment:
The function seems a bit strange, I would suggest to refactor it instead:
```c++
{
    if (m_logFile.open(QIODevice::WriteOnly | QIODevice::Append | QIODevice::Text))
        return;

    if (m_logFile.setPermissions(QFile::ReadOwner | QFile::WriteOwner)
        && m_logFile.open(QIODevice::WriteOnly | QIODevice::Append | QIODevice::Text))
        return;

    LogMsg(tr("An error occurred while trying to open the log file. Logging to file is disabled."), Log::CRITICAL);
}
```

Reply:
@Chocobo1 
Wasn't original intention to explicitly set particular permissions to log file (when it is created)? Namely unset all except read/write for owner.

Reply:
>Wasn't original intention to explicitly set particular permissions to log file (when it is created)? Namely unset all except read/write for owner.

I don't know. Do you think that operation is really needed? I'm not certain.

Reply:
> Do you think that requirement is really needed? I'm not certain.

I just noticed that this was its original intention. You can ask @sledgehammer999 about its meaning, since he is its author.
If it turns out that you really don't need it, then you can just leave these permissions alone:
```c++
{
    if (!m_logFile.open(QIODevice::WriteOnly | QIODevice::Append | QIODevice::Text)
    {
        LogMsg(tr("An error occurred while trying to open the log file. Logging to file is disabled. File: \"%1\". Error: \"%2\".")
                .arg(m_logFile.fileName(), m_logFile.errorString()), Log::CRITICAL);
    }
}
```

Reply:
>If it turns out that you really don't need it, then you can just leave these permissions alone:

I interpret it would like to try getting write permission when the first `open()` failed.